frontend/src/components/workout plan/ComboBox.tsx

```diff
@@ -9,49 +9,53 @@ import {
 } from "@/components/ui/command";
 import { Popover, PopoverContent, PopoverTrigger } from "@/components/ui/popover";
 import { Button } from "@/components/ui/button";
+import { ApiResponse } from "@/types/types";
+import Loader from "../ui/Loader";
+import { useQuery, useQueryClient } from "@tanstack/react-query";
+import { FULL_DAY_STALE_TIME, HOUR_STALE_TIME } from "@/constants/constants";
+import ErrorPage from "@/pages/ErrorPage";
+import InputSkeleton from "../ui/skeletons/InputSkeleton";
 
 interface ComboBoxProps {
   optionsEndpoint?: string;
   handleChange: (value: any) => void;
   existingValue?: string;
-  getOptions: (endpoint?: string) => Promise<any[]>;
+  getOptions: (endpoint?: any) => Promise<ApiResponse<any[]>>;
+  queryKey: string;
 }
 
 const ComboBox: React.FC<ComboBoxProps> = ({
   optionsEndpoint,
   getOptions,
   handleChange,
   existingValue,
+  queryKey,
 }) => {
   const [open, setOpen] = useState<boolean>(false);
   const [value, setValue] = useState<string | undefined>(existingValue);
-  const [values, setValues] = useState<any[]>();
+
+  const data = useQuery({
+    queryFn: () => getOptions(optionsEndpoint),
+    staleTime: FULL_DAY_STALE_TIME,
+    queryKey: [queryKey],
+  });
 
   const onChange = (val: string) => {
-    if (!values) return;
+    if (!data.data?.data) return;
```

Comment:
bruv. data.data.data is crazy

Honestly dont care too much about things like this right now though but its just funny. 

Try and make that easier to understand next time.